api-gateway/src/main/resources/static/scripts/visit-list/visit-list.controller.js

```diff
@@ -78,15 +85,28 @@ angular.module('visitList')
                     console.log(error, 'Could not receive data');
                 }
             }
+        };
 
-            function delayedReload() {
-                var loadingIndicator = document.getElementById('loadingIndicator');
-                loadingIndicator.style.display = 'block';
-                setTimeout(function() {
-                    location.reload();
-                }, 1000); //delay by 1 second
+        $scope.deleteAllCancelledVisits = function () {
+            let varIsConf = confirm('You are about to delete all canceled visits. Is this what you want to do ? ');
+            if (varIsConf) {
+                $http.delete('api/gateway/visits/cancelled')
+                    .then(successCallback, errorCallback);
+
+                function successCallback(response) {
+                    $scope.errors = [];
+                    alert("All canceled visits were deleted successfully");
+                    console.log(response, 'res');
+                    delayedReload();
+                }
+
+                function errorCallback(error) {
+                    alert(data.errors);
+                    console.log(error, 'Could not receive data');
+                }
             }
         };
+
```

Comment:
Can you explain this?

Reply:
> Can you explain this?

I know this isn't my pr, but this is the front-end method used by the button

Success callback and Error callback are the responses that are given if te function is successful or fails

Reply:
Of course bellume! So this is defining the deleteAllCancelledVisits method in the front end that we then link our button to in visit-list.template.html.  It displays a confirmation pop-up after the button press and waits for the user to confirm before actually making a request to the "visits/cancelled" uri that we defined in the backend. If this is successful it calls the successCallback method that displays a message as well as calls our delayedReload method that will refresh our data. If there would be an issue, it would display 'Could not receive  data' instead.